config/constants/settings/definition.rb

```diff
@@ -278,32 +288,80 @@ def unescape_underscores(path_segment)
         path_segment.gsub '__', '_'
       end
 
-      def env_name(definition)
+      def find_env_var_override(definition)
+        found_env_name = possible_env_names(definition).find { |name| ENV.key?(name) }
+        return unless found_env_name
+
+        if found_env_name == env_name_unprefixed(definition)
+          Rails.logger.warn(
+            "Using unprefixed environment variables is deprecated. " \
+            "Please use #{env_name(definition)} instead of #{env_name_unprefixed(definition)}"
+          )
+        end
+        yield found_env_name, ENV.fetch(found_env_name)
+      end
+
+      def each_env_var_hash_override(definition)
+        hash_override_matcher =
+          if definition.env_alias
+            /^(?:#{env_name(definition)}|#{env_name_legacy(definition)}|#{env_name_alias(definition)})_(.+)/i
+          else
+            /^(?:#{env_name(definition)}|#{env_name_legacy(definition)})_(.+)/i
+          end
+        ENV.each do |env_var_name, env_var_value|
+          env_var_name.match(hash_override_matcher) do |m|
+            yield env_var_name, env_var_value, m[1]
+          end
+        end
+      end
+
+      def possible_env_names(definition)
+        [
+          env_name_legacy(definition),
+          env_name(definition),
+          env_name_unprefixed(definition),
+          env_name_alias(definition)
+        ].compact
+      end
+
+      def env_name_legacy(definition)
```

Comment:
Since this is no longer deprecated for the time being I'd prefer this to be named differently. How about `env_name_nested`?

Reply:
good point.

I'm just seeing your comment, and Oliver already took care of it :)